lock.go

```diff
@@ -18,13 +18,29 @@ import (
 const LockName = "Gopkg.lock"
 
 type Lock struct {
-	Memo []byte
-	P    []gps.LockedProject
+	SolveMeta SolveMeta
+	P         []gps.LockedProject
+}
+
+type SolveMeta struct {
+	Memo            []byte
+	AnalyzerName    string
+	AnalyzerVersion int
+	SolverName      string
+	SolverVersion   int
 }
 
 type rawLock struct {
-	Memo     string             `toml:"memo"`
-	Projects []rawLockedProject `toml:"projects"`
+	SolveMeta solveMeta          `toml:"solve-meta"`
+	Projects  []rawLockedProject `toml:"projects"`
+}
+
+type solveMeta struct {
+	Memo            string `toml:"inputs-hash"`
```

Comment:
How do you feel about renaming Memo to InputsHash? Small nit but would make it easier on contributors if the names lined up.